src/main/scala/clump/ClumpSource.scala

```diff
@@ -16,14 +15,30 @@ class ClumpSource[T, U](val fetch: Set[T] => Future[Map[T, U]], val maxBatchSize
 
   def get(input: T): Clump[U] =
     new ClumpFetch(input, ClumpContext().fetcherFor(this))
+
+  def maxBatchSize(int: Int) =
+    new ClumpSource(fetch, maxBatchSize = int)
 }
 
 object ClumpSource {
-  def zip[T, U](fetch: List[T] => Future[List[U]], maxBatchSize: Int): ClumpSource[T, U] = {
+
+  def apply[T, U, C](fetch: C => Future[Iterable[U]])(keyExtractor: U => T)(implicit cbf: CanBuildFrom[Nothing, T, C]) =
```

Comment:
Yeah awesome. I was following Scala For The Impatient trying to figure out how to use this, [this](https://github.com/fwbrasil/clump/compare/fwbrasil:master...generify-clump.source2?expand=1#diff-7d7fc3147b6d1949991af0ff3847f8daR40) is as close as I got but it wasn't quite right :) nice to see it done properly